docs/source/breakpoints.rst

```diff
@@ -0,0 +1,122 @@
+
+The callback and hit_count properties of a Breakpoint object are useful for exactly this purpose. The following syntax is used to set a callback function for your breakpoint:
+
+.. code-block:: python
+
+    def on_breakpoint_hit(d, bp):
+        print(f"RAX: {d.regs.rax}")
+
+    d.breakpoint(0x11f0, callback=on_breakpoint_hit)
+
+The signature of a callback function is as follows:
+
+.. code-block:: python
+
+    def callback(d: ThreadContext, bp: Breakpoint) -> None:
+
+The first parameter can either be a debugger object or a thread context object. This kind of object is described in :doc:`multithreading`.
+The second parameter is the breakpoint object that triggered the callback.
+
+As for the hit_count property, the following is an example of how to it:
+
+.. code-block:: python
+
+    while br.hit_count < 100:
+        d.cont()
+        print(f"Hit count: {d.hit_count}")
+
+
+Symbolic addressing
+^^^^^^^^^^^^^^^^^^^
+
+Just like with memory access, you can use symbolic addressing to set breakpoints. The following syntax is used to set a breakpoint at a specific function:
+
+.. code-block:: python
+
+    d.breakpoint("vuln")
+
+Relative addressing with respect to a symbol is also supported. Just like in GDB, the offset is specified in decimal format:
+
+.. code-block:: python
+
+    d.breakpoint("vuln+10")
+
+Hardware breakpoints
+^^^^^^^^^^^^^^^^^^^^
+
+You can easily set a hardware breakpoint with the same api as a software breakpoint. Just set the hardware parameter to True:
+
+.. code-block:: python
+
+    d.breakpoint(0x10ab, hardware=True)
+
+As previously mentioned, hardware breakpoints are limited in number. For example, in the x86 architecture, there are only 4 hardware breakpoints available. If you exceed that number, a `RuntimeError` will be raised.
+
+Watchpoints
+-----------
+
+Watchpoints are a special type of hardware breakpoint that triggers when a specific memory location is accessed. You can set a watchpoint to trigger on read, write, read/write, or execute access.
+
+Features of watchpoints are shared with breakpoints, so you can set callbacks, check the `hit_count` and activate / deactivate the watchpoint in the same way. While you can use the breakpoint API to set up a breakpoint, a specific API is provided on watchpoints for your convenience:
+
+.. code-block:: python
+
+    def watchpoint(
+        position=...,
+        condition=...,
+        length=...,
+        callback=...) -> Breakpoint:
+
+Again, the position can be specified both as a relative address or as a symbol.
+The condition parameter specifies the type of access that triggers the watchpoint. The following values are supported:
+
+- ``"r"``: read access
+- ``"w"``: write access
+- ``"rw"``: read/write access
+- ``"x"``: execute access
+
+By default, the watchpoint is triggered only on write access.
+
+The length parameter specifies the size of the word being watched. The following values are supported:
+
+- ``1``: byte
```

Comment:
This is correct for the RISC-V architecture where a dword is 8 bytes, but not for x86_64.